File: client/src/types/bannercourse.ts

export interface BannerCourseResponse {
    bannerCourseId: string;
    title: string;
    description: string;
    imageUrl: string;
}

export interface CreateBannerCourseRequest {
    title: string;
    description: string;
    image: File;
}

export interface UpdateBannerCourseRequest {
    title?: string;
    description?: string;
    image?: File;
}

File: server/src/main/java/vn/edu/actvn/server/entity/BannerCourse.java
package vn.edu.actvn.server.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class BannerCourse {
    @Id @GeneratedValue(strategy = GenerationType.UUID)
    String bannerCourseId;
    @Column(nullable = false, length = 100)
    String title;
    String description;
    @Column(nullable = false, length = 500)
    String imageUrl;
    String publicId;
}
